src/common/atLeastOne.spec.ts

```diff
@@ -0,0 +1,17 @@
+import { assertType } from 'vitest'
+import type { AtLeastOne } from './atLeastOne'
+
+describe('AtLeastOne', () => {
+  it('makes at least one key of an object required', () => {
+    interface TestType {
+      a: string
+      b: number
+    }
+
+    const testCase1: AtLeastOne<TestType> = { a: 'a' }
+    const testCase2: AtLeastOne<TestType> = { b: 1 }
+
+    assertType<{ a: string; b?: number }>(testCase1)
+    assertType<{ a?: string; b: number }>(testCase2)
```

Comment:
can you also add negative case?

Reply:
Added negative test ✅ 

I've also enabled `typecheck` as by default vitest has it disabled - please see `vitest.config.mts`